.github/workflows/preview-env-teardown.yml

```diff
@@ -32,6 +32,25 @@ jobs:
       - name: Set up Cloud SDK
         uses: google-github-actions/setup-gcloud@v2
 
+      - name: Find deployment status comment
+        uses: peter-evans/find-comment@v3
+        id: find-comment
+        with:
+          issue-number: ${{ github.event.number }}
+          body-includes: <!-- preview-env -->
+
+      - name: Shorten commit SHA
+        run: echo "short_sha=$(echo ${{ github.sha }} | cut -c1-8)" >> $GITHUB_ENV
+
+      - name: Update comment with tear-down warning
+        if: steps.find-comment.outputs.comment-id != ''
+        uses: peter-evans/create-or-update-comment@v4
+        with:
```

Comment:
this step needs to be moved above, before the `Set up Cloud SDK` step, as that one takes quite some time (the idea is to notify the user of the deleting process as early as possible, here we're waiting for the GCloud to be installed and that can take a minute)